build/dsc/templates/dsc_type.rb.erb

```diff
@@ -4,6 +4,7 @@ Puppet::Type.newtype(:dsc_<%= resource.friendlyname.downcase %>) do
   require Pathname.new(__FILE__).dirname + '../../' + 'puppet/type/base_dsc'
 
   provide :powershell, :parent => Puppet::Type.type(:base_dsc).provider(:powershell) do
+    confine :operatingsystem => :windows, :powershell_version => '5.0.10514.6'
```

Comment:
I'm not quite sure how that would work without custom comparisons, given
that's not a version object.

Something like http://stackoverflow.com/a/3064161 should work though
On Sep 22, 2015 7:37 PM, "James Pogran" <notifications@github.com> wrote:

> In build/dsc/templates/dsc_type.rb.erb
> <https://github.com/puppetlabs/puppetlabs-dsc/pull/99#discussion_r40166375>
> :
>
> > @@ -4,6 +4,7 @@ Puppet::Type.newtype(:dsc_<%= resource.friendlyname.downcase %>) do
> >    require Pathname.new(__FILE__).dirname + '../../' + 'puppet/type/base_dsc'
> >
> >    provide :powershell, :parent => Puppet::Type.type(:base_dsc).provider(:powershell) do
> > +    confine :operatingsystem => :windows, :powershell_version => '5.0.10514.6'
>
> This breaks all tests.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/puppetlabs/puppetlabs-dsc/pull/99/files#r40166375>.
>